src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -171,6 +175,27 @@ def prefork(self, options, build_config):
       self._watchman_is_running.wait()
 
     session = self._graph_helper.new_session()
+    if options.for_global_scope().loop:
+      return session, self._prefork_loop(session, options)
+    else:
+      return session, self._prefork_body(session, options)
+
+  def _prefork_loop(self, session, options):
+    iterations = options.for_global_scope().loop_max
+    target_roots = None
+    while iterations and not self.is_killed:
+      try:
+        target_roots = self._prefork_body(session, options)
+      except session.scheduler_session.execution_error_type as e:
```

Comment:
This feels like a pretty major behaviour change? Can you talk through a little why raising -> not raising is safe?

Reply:
Isn't this because the whole point of looping is to just display errors but continue to allow for a live edit to fix the error(s)?

Reply:
Yep, that.